src/components/ContactForm.js

```diff
@@ -10,18 +10,21 @@ const propTypes = {
     name: PropTypes.string,
     phoneNumber: PropTypes.string,
   })).isRequired,
-  onAddContact: PropTypes.func.isRequired,
+  onUpdateContact: PropTypes.func.isRequired,
 };
 
 class ContactForm extends Component {
   constructor(props) {
     super(props);
 
+    this.isEditMode = this.props.match.path.includes('/edit-contact/');
+    this.contactId = parseInt(this.props.match.params.contactId);
```

Comment:
The same, you are duplicate data

Reply:
replace contactId property with getContactId method